apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/FissClaimRdaSink.java

```diff
@@ -0,0 +1,128 @@
+package gov.cms.bfd.pipeline.rda.grpc.sink;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.zaxxer.hikari.HikariDataSource;
+import gov.cms.bfd.model.rda.PreAdjFissClaim;
+import gov.cms.bfd.pipeline.rda.grpc.ProcessingException;
+import gov.cms.bfd.pipeline.rda.grpc.RdaSink;
+import gov.cms.bfd.pipeline.sharedutils.DatabaseOptions;
+import gov.cms.bfd.pipeline.sharedutils.DatabaseUtils;
+import java.util.Collection;
+import javax.persistence.EntityExistsException;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** RdaSink implementation that writes PreAdjFissClaim objects to the database in batches. */
+public class FissClaimRdaSink implements RdaSink<PreAdjFissClaim> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(FissClaimRdaSink.class);
+  public static final String CALLS_METER_NAME =
+      MetricRegistry.name(FissClaimRdaSink.class.getSimpleName(), "calls");
+  public static final String FAILURES_METER_NAME =
+      MetricRegistry.name(FissClaimRdaSink.class.getSimpleName(), "failures");
+  public static final String PERSISTS_METER_NAME =
+      MetricRegistry.name(FissClaimRdaSink.class.getSimpleName(), "persists");
+  public static final String MERGES_METER_NAME =
+      MetricRegistry.name(FissClaimRdaSink.class.getSimpleName(), "merges");
+
+  private final HikariDataSource dataSource;
+  private final EntityManagerFactory entityManagerFactory;
+  private final EntityManager entityManager;
+  private final Meter callsMeter;
+  private final Meter failuresMeter;
+  private final Meter persistsMeter;
+  private final Meter mergesMeter;
+
+  public FissClaimRdaSink(DatabaseOptions databaseOptions, MetricRegistry metricRegistry) {
+    dataSource = DatabaseUtils.createDataSource(databaseOptions, metricRegistry, 10);
+    entityManagerFactory = DatabaseUtils.createEntityManagerFactory(dataSource);
+    entityManager = entityManagerFactory.createEntityManager();
+    callsMeter = metricRegistry.meter(CALLS_METER_NAME);
+    failuresMeter = metricRegistry.meter(FAILURES_METER_NAME);
+    persistsMeter = metricRegistry.meter(PERSISTS_METER_NAME);
+    mergesMeter = metricRegistry.meter(MERGES_METER_NAME);
+  }
+
+  @Override
+  public void close() throws Exception {
+    entityManager.close();
+    entityManagerFactory.close();
+    dataSource.close();
+  }
+
+  /**
+   * Writes the entire batch to the database in a single transaction.
+   *
+   * @param claims batch of claims to be written to the database
+   * @return the number of claims successfully written to the database
+   * @throws ProcessingException wrapped exception if an error takes place
+   */
+  @Override
+  public int writeBatch(Collection<PreAdjFissClaim> claims) throws ProcessingException {
```

Comment:
The logic in this class has decent complexity, do we want to create unit tests for this?